### PR TITLE
[vulkan] Add a doc to describe generic Vulkan environment setup

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: docs/GetStarted/generic_vulkan_env_setup.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+* (NVIDIA) [nvidia-vulkan-icd][PackageNvidiaVulkan] for NVIDIA Vulkan ICD.
+
+The above packages provide the Vulkan loader and ICDs. With them an Vulkan
+application should be able to run. You may additionally want to install
+
+* [vulkan-tools][PackageVulkanTools] for command-line tools like `vulkaninfo`
+  (dumping available ICDs and their capabilities) and GUI application like
+  `vulkancube` (rendering a rotating cube).
+
+To test the runtime functions properly.
```
Comment (ScottTodd) [inline docs/GetStarted/generic_vulkan_env_setup.md:62]: nit: sentence fragment, could lowercase 'to' so this continues the previous fragment?

File: docs/GetStarted/generic_vulkan_env_setup.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+### Debian/Ubuntu
+
+The following packages should be installed for a proper Vulkan runtime:
+
+* [libvulkan1][PackageLibVulkan1] for the Vulkan loader `libvulkan.so`.
+* (AMD) [mesa-vulkan-drivers][PackageMesaVulkan] for Mesa AMD Vulkan ICD.
+* (NVIDIA) [nvidia-vulkan-icd][PackageNvidiaVulkan] for NVIDIA Vulkan ICD.
+
+The above packages provide the Vulkan loader and ICDs. With them an Vulkan
+application should be able to run. You may additionally want to install
```
Comment (ScottTodd) [inline docs/GetStarted/generic_vulkan_env_setup.md:56]: Can we recommend a specific version of the loader? See https://github.com/google/iree/issues/1576

Comment (antiagainst) [reply docs/GetStarted/generic_vulkan_env_setup.md:57]: Good point. But this doc is meant to be generic without IREE specific things. I think we should put it in the IREE specific docs. Updated there.

File: docs/GetStarted/getting_started_linux_vulkan.md [hunk 3]
```diff
@@ -123,6 +123,13 @@ $ export VK_LAYER_PATH=$PWD/build/third_party/vulkan_extensionlayer/layers/:$VK_
 $ export VK_LAYER_PATH=$PWD/bazel-bin/external/vulkan_extensionlayer/:$VK_LAYER_PATH
 ```
 
+### Setting up the Vulkan Loader
+
+IREE relies on the `VK_KHR_timeline_semaphore` extension. The minimal loader
+version recognizes this extenion is `1.1.124`. So if you see failures regarding
```
Comment (ScottTodd) [inline docs/GetStarted/getting_started_linux_vulkan.md:129]: nit: recognizes -> recognizing (or "supporting")

**Pull request closed without merging**